app/tests/test_collections_endpoint.py

```diff
@@ -240,3 +255,39 @@ def test_collection_patch(self):
 
         # description not affected by patch
         self.assertEqual(self.collections[1].description, response_json['description'])
+
+    def test_unauthorized_collection_post_put_patch(self):
+        # make sure POST fails for anonymous user:
+        # a post with the absolute valid minimum
+        payload_json = self.sample_collections['valid_min_collection_set_1']
+
+        response = self.client.post(
+            f"/{API_BASE}/collections", data=payload_json, content_type='application/json'
+        )
+        self.assertEqual(401, response.status_code, msg="Unauthorized post was permitted.")
+
+        # make sure PUT fails for anonymous user:
+        payload_json = self.sample_collections['valid_collection_set_3']
+        # for the start, the id have to be different
+        self.assertNotEqual(self.collections[0].name, payload_json['id'])
+        response = self.client.put(
+            f"/{API_BASE}/collections/{self.collections[0].name}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        self.assertEqual(401, response.status_code, msg="Unauthorized put was permitted.")
+
+        # make sure PATCH fails for anonymous user:
+        collection_name = self.collections[1].name  # get a name that is registered in the service
+        payload_json = self.sample_collections['less_than_min_collection_set']
+        payload_json['id'] = collection_name  # rename the payload to this name
+        # for the start, the collection[1] has to have a different licence than the payload
+        self.assertNotEqual(self.collections[1].license, payload_json['license'])
+        # for start the payload has no description
+        self.assertNotIn('title', payload_json.keys())
+        response = self.client.patch(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        self.assertEqual(401, response.status_code, msg="Unauthorized patch was permitted.")
```

Comment:
I would also add the `DEL` case that should return 404 (because it not implemented) with a comment that if it gets implemented then it needs to return 401. This way we are sure that if DEL gets in future implemented we don't forget to have it also tested against  login.

Reply:
Good point. Added tests for authorized and unauthorized delete requests.